hudi-examples/hudi-examples-spark/src/test/python/HoodiePySparkQuickstart.py

```diff
@@ -0,0 +1,260 @@
+
+        # issue deletes
+        hudi_hard_delete_options = {
+            'hoodie.table.name': self.tableName,
+            'hoodie.datasource.write.recordkey.field': 'uuid',
+            'hoodie.datasource.write.partitionpath.field': 'partitionpath',
+            'hoodie.datasource.write.table.name': self.tableName,
+            'hoodie.datasource.write.operation': 'delete',
+            'hoodie.datasource.write.precombine.field': 'ts',
+            'hoodie.upsert.shuffle.parallelism': 2, 
+            'hoodie.insert.shuffle.parallelism': 2
+        }
+
+        deletes = list(map(lambda row: (row[0], row[1]), ds.collect()))
+        hard_delete_df = self.spark.sparkContext.parallelize(deletes).toDF(['uuid', 'partitionpath']).withColumn('ts', lit(0.0))
+        hard_delete_df.write.format("hudi").options(**hudi_hard_delete_options).mode("append").save(self.basePath)
+
+        # run the same read query as above.
+        roAfterDeleteViewDF = self.spark.read.format("hudi").load(self.basePath) 
+        roAfterDeleteViewDF.createOrReplaceTempView("hudi_trips_snapshot")
+        # fetch should return (total - 2) records
+        total_count = self.spark.sql("SELECT uuid, partitionpath FROM hudi_trips_snapshot").count()
+        print(f"total count: {total_count}")
+        return hard_delete_df
+
+    def insertOverwrite(self):
+        print("Insert Overwrite")
+        self.spark.read.format("hudi").load(self.basePath).select(["uuid","partitionpath"]).sort(["partitionpath", "uuid"]).show(n=100,truncate=False)
+        inserts = self.spark._jvm.org.apache.hudi.QuickstartUtils.convertToStringList(self.dataGen.generateInserts(10))
+        df = self.spark.read.json(self.spark.sparkContext.parallelize(inserts, 2)).filter("partitionpath = 'americas/united_states/san_francisco'")
+        hudi_insert_overwrite_options = {
+            'hoodie.table.name': self.tableName,
+            'hoodie.datasource.write.recordkey.field': 'uuid',
+            'hoodie.datasource.write.partitionpath.field': 'partitionpath',
+            'hoodie.datasource.write.table.name': self.tableName,
+            'hoodie.datasource.write.operation': 'insert_overwrite',
+            'hoodie.datasource.write.precombine.field': 'ts',
+            'hoodie.upsert.shuffle.parallelism': 2,
+            'hoodie.insert.shuffle.parallelism': 2
+        }
+        df.write.format("hudi").options(**hudi_insert_overwrite_options).mode("append").save(self.basePath)
+        self.spark.read.format("hudi").load(self.basePath).select(["uuid","partitionpath"]).sort(["partitionpath", "uuid"]).show(n=100,truncate=False)
+        return df
+
+if __name__ == "__main__":
+    random.seed(46474747)
+    if len(sys.argv) < 3:
+        print("Usage: python3 HoodiePySparkQuickstart.py <tableName> <jar file path/bundle name>")
+        quit(-1)
+    #Example jar filepath: /Users/jon/.m2/repository/org/apache/hudi/hudi-spark3.3-bundle_2.12/0.13.0-SNAPSHOT/hudi-spark3.3-bundle_2.12-0.13.0-SNAPSHOT.jar
```

Comment:
Example jar path: [HUDI_BASE_PATH]/packaging/hudi-spark-bundle/target/hudi-spark-bundle[VERSION].jar 
do not add any absolute paths in general.